File: new/atoms/stage-label.test.mjs
import { stageLabel as __pure_stageLabel } from './stage-label.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הטיהור v3)
const __d_stageLabel_STAGE_LABEL_T = {
  k1: "חדש",
  k2: "בהכנה",
  k3: "רישום",
  k4: "מסירה",
  k5: "הושלם",
  k6: "ayin.stage.",
};
const stageLabel = (...a) => __pure_stageLabel(...a, ...Array(Math.max(0, 3 - a.length)).fill(undefined), __d_stageLabel_STAGE_LABEL_T);
let f = 0;
const eq = (a, b, msg) => { if (a !== b) { console.error(`✗ ${msg} ⇒ ${JSON.stringify(a)}`); f = 1; } };

// 1) מונח-ארגון גובר
eq(stageLabel({}, 'eyes', () => 'ביצוע'), 'ביצוע', 'מונח-הארגון לא גבר');

// 2) נפילה לברירת-המחדל
const termOf = (c, k, fb) => c.terms?.[k] ?? fb;
eq(stageLabel({}, 'answer', termOf), 'מסירה', 'ברירת-המחדל של answer שגויה');

// 3) חמשת השלבים בנפילה
eq(stageLabel({}, 'new', termOf), 'חדש', 'new שגוי');
eq(stageLabel({}, 'lead', termOf), 'בהכנה', 'lead שגוי');
eq(stageLabel({}, 'eyes', termOf), 'רישום', 'eyes שגוי');
eq(stageLabel({}, 'done', termOf), 'הושלם', 'done שגוי');

// 4) השקע נקרא פעם אחת עם (cfg, 'ayin.stage.done', 'הושלם')
const cfg = { tag: 'cfg' };
let calls = 0, got = null;
stageLabel(cfg, 'done', (...args) => { calls++; got = args; return 'x'; });
if (calls !== 1 || got[0] !== cfg || got[1] !== 'ayin.stage.done' || got[2] !== 'הושלם') {
  console.error(`✗ קריאת-השקע שגויה ⇒ ${JSON.stringify(got)}`); f = 1;
}

if (f) process.exit(1);
console.log('✓ stage-label: 4 דוגמאות-חוזה — ירוק');

File: new/dart-maor/stage-label_test.dart
import '../dart-data-maor/stage-label.dart';
// בדיקת-חוזה (רתמת-זהב) · stageLabel — מייבאת אך ורק את האטום-שלה (חוק-4).
// מתרגמת את כל 4 דוגמאות-החוזה (stage-label.contract.md) ואת בדיקת-ה-JS
// (new/atoms/stage-label.test.mjs) ביט-אחר-ביט: מונח-ארגון גובר / נפילה /
// חמשת-השלבים / קריאת-שקע יחידה עם (cfg, 'ayin.stage.done', 'הושלם') —
// אותו אובייקט-cfg כלשונו (identical). כשל ⇒ StateError. אם עובר ⇒ Dart≡JS.
// הרצה: dart run --enable-asserts new/dart-maor/stage-label_test.dart ⇒ exit 0
import 'stage-label.dart';

void _ok(bool cond, String msg) {
  if (!cond) throw StateError('FAIL: $msg');
}

/// שקע-termOf נופל כמו בבדיקת-ה-JS: (c,k,fb) => c.terms?.[k] ?? fb.
dynamic _termOf(dynamic c, dynamic k, dynamic fb) {
  final terms = (c is Map) ? c['terms'] : null;
  final v = (terms is Map) ? terms[k] : null;
  return v ?? fb;
}

void main() {
  var n = 0;

  // 1) מונח-ארגון גובר — השקע מחזיר 'ביצוע' (ורטיקל הסטודיו).
  {
    final got = stageLabel(<String, dynamic>{}, 'eyes', (c, k, fb) => 'ביצוע', stageFallback: kStageFallback);
    _ok(got == 'ביצוע', "מונח-הארגון לא גבר (בפועל: $got)");
    n++;
  }

  // 2) נפילה לברירת-המחדל — cfg ריק ⇒ 'מסירה'.
  {
    final got = stageLabel(<String, dynamic>{}, 'answer', _termOf, stageFallback: kStageFallback);
    _ok(got == 'מסירה', "ברירת-המחדל של answer שגויה (בפועל: $got)");
    n++;
  }

  // 3) חמשת השלבים בנפילה.
  {
    final expected = <String, String>{
      'new': 'חדש',
      'lead': 'בהכנה',
      'eyes': 'רישום',
      'answer': 'מסירה',
      'done': 'הושלם',
    };
    expected.forEach((stage, want) {
      final got = stageLabel(<String, dynamic>{}, stage, _termOf, stageFallback: kStageFallback);
      _ok(got == want, "$stage שגוי (בפועל: $got ≠ $want)");
    });
    n++;
  }

  // 4) השקע נקרא בדיוק פעם אחת עם (cfg, 'ayin.stage.done', 'הושלם') —
  //    אותו אובייקט-cfg כלשונו (identical, כמו !== ב-JS).
  {
    final cfg = <String, dynamic>{'tag': 'cfg'};
    var calls = 0;
    List<dynamic>? got;
    stageLabel(cfg, 'done', (c, k, fb) {
      calls++;
      got = [c, k, fb];
      return 'x';
    }, stageFallback: kStageFallback);
    _ok(
        calls == 1 &&
            got != null &&
            identical(got![0], cfg) &&
            got![1] == 'ayin.stage.done' &&
            got![2] == 'הושלם',
        'קריאת-השקע שגויה (calls=$calls, args=$got)');
    n++;
  }

  print('OK stageLabel: $n contract examples passed');
}
